### ISSUE TITLE
Missing symbolic link to data run (Trac #2201)

### ISSUE THREAD

**Issue by icecube-trac** [icecube-trac/tix4#2193]
Hi,

there is a symbolic link missing for run 00131184.
/data/exp/IceCube/2018/filtered/level2/0619/Run00131184_73
does exist but
/data/exp/IceCube/2018/filtered/level2/0619/Run00131184
does not.

That would not be too annoying but the goodRun file references the /data/exp/IceCube/2018/filtered/level2/0619/Run00131184 folder. 
As the other folders for 2018 seem to be mostly symlinked, I would like this folder to be symlinked as well.

Cheers,
Frederik

<details>
 <summary><em>Migrated from <a href="https://code.icecube.wisc.edu/projects/icecube/ticket/2201">https://code.icecube.wisc.edu/projects/icecube/ticket/2201</a>, reported by flauberand owned by rsnihur</em></summary>
<p>

```json
{
    "status": "closed",
    "changetime": "2019-02-13T14:18:07",
    "_ts": "1550067487593768",
    "description": "Hi,\n\nthere is a symbolic link missing for run 00131184.\n/data/exp/IceCube/2018/filtered/level2/0619/Run00131184_73\ndoes exist but\n/data/exp/IceCube/2018/filtered/level2/0619/Run00131184\ndoes not.\n\nThat would not be too annoying but the goodRun file references the /data/exp/IceCube/2018/filtered/level2/0619/Run00131184 folder. \nAs the other folders for 2018 seem to be mostly symlinked, I would like this folder to be symlinked as well.\n\nCheers,\nFrederik\n\n",
    "reporter": "flauber",
    "cc": "",
    "resolution": "wontfix",
    "time": "2018-10-29T17:24:29",
    "component": "infrastructure",
    "summary": "Missing symbolic link to data run",
    "priority": "normal",
    "keywords": "",
    "milestone": "",
    "owner": "rsnihur",
    "type": "defect"
}
```

</p>
</details>

**Comment by icecube-trac** [comment]
Trac update at 20181029T18:17:49: 
* olivas changed status from "new" to "assigned"
* olivas changed owner from "" to "rsnihur"

**Comment by icecube-trac** [comment]
Trac update at 20181031T15:34:46: rsnihur commented:

> Hi Frederik,
> Below I cut-and-pasted 2 discussions about this issue from the Slack
> #data-processing channel dated June 25-26 & September 11.
> 
> AFAIK the test runs should be in IC86_2018_GoodRunInfo.txt 
> 
> Thx,
> -Rob
> 
> 
> -----
> 
> flambda [12:21 PM]
> btw, will the IC86_2018_GoodRunInfo.txt be fixed? It actually does point to non existent folders (my guess is someone forgot to add the symlinks to the _73 folders)
> 
> flambda [8:12 AM]
> and while we are at it, should the 2017 test runs also show up in the IC86_2017_GoodRunInfo? I have been pooling these files for infos on hte runs to not manually walk over the fs, is there a better way to do that?
> 
> Rob Snihur [9:06 AM]
> I can take a look at the IC86_2018_GoodRunInfo.txt , but should the test runs actually be in there?  Especially if there are no test runs in the IC86_2017_GoodRunInfo.txt ...
> 
> dschultz [9:10 AM]
> I think if the test runs are marked as GOOD they should be in the good run list.  for 2017, I think they should be present.  for 2018, they probably need to be approved first
> 
> jkelley [9:24 AM]
> FYI, the 2017 test runs (129393–129397) are marked BAD.
> As @dschultz says, good test runs should be included in the good run list the year they are the test for
> 
> flambda [9:36 AM]
> hm, so what is the canonical way to poll information about runs? Especially since we have pass2 and if I also want to know there the data is stored? In the past I just got me a local copy of the https://live.icecube.wisc.edu/snapshots but that afaik does not give the path to the data (also, maybe I should move this to software)
> 
> jkelley [9:42 AM]
> As far as I know the good run list file is intended to be the canonical way to provide paths to the L2 data
> 
> flambda [9:44 AM]
> kk, thanks
> 
> -------
> 
> 
> 
> thomas [4:46 AM]
> Hi!
> I was looking for 2018 Level2 data and cannot find it… The folders in `/data/exp/IceCube/2018/filtered/level2` stop on July 8th, the date of the switch to 2018 data-taking. Also, the corresponding GRL only mentions the 2018 test runs (`/data/exp/IceCube/2018/filtered/level2/IC86_2018_GoodRunInfo.txt`). (edited)
> Also, the symlinks for the runs (e.g. `Run00131185_73/` to `Run00131185/`) are missing.
> 
> nick [6:05 AM]
> joined #data-processing.
> 
> dschultz [8:02 AM]
> they're not processed yet
> I think 2018 L2 will be starting soon, and you'll hear more about it at the collaboration meeting
> 
> Rob Snihur [9:38 AM]
> Hi @thomas I am working on it
> 
> Rob Snihur [11:40 AM]
> @thomas NB: The symlinks are purposely not created for test runs (131185 is a test run)
> 
> thomas [11:40 AM]
> thanks for looking into it, Rob!
> but the test runs are mentioned in the GoodRunInfo.txt file and the entry there points to the (missing) symlink (edited)
> 
> Rob Snihur [11:45 AM]
> I'm not sure why the GoodRunInfo.txt file points to the missing symlink.  Can you check if that's how it also was for the 2017 season?  I know that the L2 processing scripts explicitly do not create the links for test runs
> 
> thomas [4:10 PM]
> The last time that the test runs were included was in 2015, the GoodRunInfo.txt back then points to the symlinks (which exist).
> I also checked the GoodRunInfo_Versioned.txt, which should point to the correct folders, but it points to e.g. `Run00131184_74` when the folder is actually `Run00131184_73`.
> For 2016 and 2017, the symlinks are not there, but the runs also do not show up in the GoodRunInfo.txt. I think this might be the actual issue, as the entries are also incomplete (having `N/A` for `ActiveStrings`, etc.). There is one more run in the file which is not a test run, and everything looks good there.
> I just noticed this because my parsing script from the previous years failed on the 2018 GoodRunInfo.txt file.

**Comment by icecube-trac** [comment]
Trac update at 20190213T14:18:07: 
* olivas changed status from "assigned" to "closed"
* olivas changed resolution from "" to "wontfix"
